haskell/private/haskell_impl.bzl

```diff
@@ -362,7 +362,7 @@ def haskell_library_impl(ctx):
     srcs_files, import_dir_map = _prepare_srcs(ctx.attr.srcs)
 
     with_shared = not ctx.attr.linkstatic
-    if with_profiling or hs.toolchain.is_static:
+    if with_profiling or hs.toolchain.static_runtime:
```

Comment:
I don't _think_ it'll be an issue here based on previous comments, but if we ever _do_ want a dynamic runtime with a static binary set (honestly I still don't think I understand how this could work 😂 ) this will need to be unpicked/tested I think.